more_itertools/more.py

```diff
@@ -1004,3 +1004,53 @@ def divide(n, iterable):
         ret.append(iter(seq[start:stop]))
 
     return ret
+
+
+def always_iterable(obj):
+    """
+    Given an object, always return an iterable.
+
+    If the object is not already iterable, return a tuple containing containing
+    the object::
+
+        >>> always_iterable(1)
+        (1,)
+
+    If the object is ``None``, return an empty iterable::
+
+        >>> always_iterable(None)
+        ()
+
+    Otherwise, return the object itself::
+
+        >>> always_iterable([1, 2, 3])
+        [1, 2, 3]
+
+    Strings (binary or unicode) are not considered to be iterable::
+
+        >>> always_iterable('foo')
+        ('foo',)
+
+    This function is useful in applications where a passed parameter may be
+    either a single item or a collection of items::
+
+        >>> def item_sum(param):
+        ...     total = 0
+        ...     for item in always_iterable(param):
+        ...         total += item
+        ...
+        ...     return total
+        >>> item_sum(10)
+        10
+        >>> item_sum([10, 20])
+        30
+
+    """
+    if obj is None:
+        return ()
+
+    string_like_types = (text_type, binary_type)
```

Comment:
We had [some discussion](https://github.com/erikrose/more-itertools/pull/37#issuecomment-39687015) about this way back when. I've got this using `text_type` and `binary_type` here, so we'll get `(unicode, str)` on Python 2 and `(str, bytes)` on Python 3.